src/hotspot/share/utilities/globalDefinitions.hpp

```diff
@@ -1334,6 +1334,11 @@ template<typename K> bool primitive_equals(const K& k0, const K& k1) {
   return k0 == k1;
 }
 
+template<typename K> int primitive_compare(const K& k0, const K& k1) {
+  return (((uintptr_t)k0 < (uintptr_t)k1) ? -1
+       : ((uintptr_t)k0 == (uintptr_t)k1) ? 0 : 1);
+}
+
```

Comment:
I was thinking this would hurt performance, but on x86 and aarch64 the compiler generates conditional moves instead of conditional branches.
I don't like using uintptr_t here.  Can't we compare void* or better yet K*?

Reply:
To match primitive_equals, we can just use k0 < k1, without the cast to uintptr_t, right?

Reply:
It compiles and passes the appcds tests so I'm surprised it doesn't need the casts.  I didn't think InstanceKlass pointers could be compared like that directly.

Reply:
Yes, comparing pointers directly should be fine.